lib/db/observed.go

```diff
@@ -28,11 +28,7 @@ func (db *Lowlevel) AddOrUpdatePendingDevice(device protocol.DeviceID, name, add
 
 func (db *Lowlevel) RemovePendingDevice(device protocol.DeviceID) error {
 	key := db.keyer.GeneratePendingDeviceKey(nil, device[:])
-	if err := db.Delete(key); err != nil {
-		l.Warnf("Failed to remove pending device entry: %v", err)
```

Comment:
Where? I didn't on purpose in cases where the error is passed on to the API.

Reply:
Yes it's passed on to the API, but not logged anywhere.

Reply:
Yep, and from there it's given to the UI. Indeed checking #7712 the UI seems to drop any errors there. That may be appropriate or not, but it doesn't change that the error is passed on and thus the api callers responsibility. If what you are after is debug logging: That should indeed live in observed.go (due to facilities) but that's not the intent of this PR.

Reply:
What you describe as debug logging is probably more like what I'm after.  However, in case of an unexpected (and inexplicable) failure, I don't think it helps much having to enable debug logging first.  After doing that, the error might not be reproducible, or the user does not have the patience to try, or it might be overlooked completely.  If there is a slight feeling that something didn't work as expected, one could look at the log to confirm there was a corresponding failure message.  Debug level logging IMO should be stuff that happens regularly and is just too noisy, therefore must be able to be suppressed.

Just saying that because I put the messages there with a clear intention, to help in diagnosing any problems with the code I contributed.  As noted earlier, I do intend to stick around and take responsibility for it, so I find it a bit unfortunate that my justified views were just overruled here.  Let's hope we can reach more common ground in the future.